### DISCUSSION TITLE
Guide on using TailwindCSS in Play Framework (without npm)

### DISCUSSION THREAD

**Discussion by hertg** [playframework/playframework#12835] [Show and tell]
I have just published a guide on how to use TailwindCSS in a Play Framework project, without relying on nodejs and npm. 
Since TailwindCSS offers a [standalone CLI](https://tailwindcss.com/blog/standalone-cli), and thanks to the source file tasks of [sbt-web](https://github.com/sbt/sbt-web), we can integrate the Tailwind compilation into the sbt build of Play. 

The guide can be found here: 
https://hertig.blog/posts/tailwindcss-in-play-framework-without-npm/

And an example project is publicly available here:
https://codeberg.org/hertg/play-tailwindcss

---

Maybe this will be useful for anyone who only dares to touch Javascript bundlers with a long stick... Because I haven't found a stick of sufficient length yet :)

**Comment by tanin47** [comment]
I had to write an sbt-postcss plugin for it 🤣 . Your way is much more lightweight. Thank you for sharing.
